backend/src/extension.ts

```diff
@@ -20,19 +21,27 @@ export function activate(context: vscode.ExtensionContext) {
 		return;
 	}
 
+	const inTheia = await isInTheia();
+
 	// YeomanUIPanel
-	yeomanUIPanel = new YeomanUIPanel(extContext);
+	yeomanUIPanel = new YeomanUIPanel(extContext, inTheia);
 	registerAndSubscribeCommand("loadYeomanUI", yeomanUIPanel.loadWebviewPanel.bind(yeomanUIPanel));
 	registerAndSubscribeCommand("yeomanUI.toggleOutput", yeomanUIPanel.toggleOutput.bind(yeomanUIPanel));
 	registerAndSubscribeCommand("yeomanUI._notifyGeneratorsChange", yeomanUIPanel.notifyGeneratorsChange.bind(yeomanUIPanel));
 	registerWebviewPanelSerializer(yeomanUIPanel);
 
 	// ExploreGensPanel
-	exploreGensPanel = new ExploreGensPanel(extContext);
+	exploreGensPanel = new ExploreGensPanel(extContext, inTheia);
 	registerAndSubscribeCommand("exploreGenerators", exploreGensPanel.loadWebviewPanel.bind(exploreGensPanel));
 	registerWebviewPanelSerializer(exploreGensPanel);
 }
 
+async function isInTheia() {
+	const commands = await vscode.commands.getCommands(true);
+	const foundCommands = _.intersection(commands, ["theia.open", "preferences:open", "keymaps:open", "workspace:openRecent"]);
+	return !_.isEmpty(foundCommands);
+}
+
```

Comment:
We had 2 months ago class named Theia.ts
https://github.com/SAP/yeoman-ui/commit/d109bea2bc7c9298683fee9bac5b3895da241851#diff-cf07a938a7f1b68c87573b5872d099b1

Reply:
We need to calculate it only once